### ISSUE TITLE
Process "Gathering songs..." is endless when adding a folder to a playlist

### ISSUE THREAD

**Issue by DarkSkyGhost** [timusus/Shuttle#113]
**Shuttle version:**
 
 v1.6.7 (Upgraded)
 
**Device, OS:**
 
7.1.2 LineageOS 14.1-20170726-NIGHTLY-kenzo, Xiaomi Redmi Note 3 Pro International Edition Kate

**Description of bug:**
 
When adding folders to a new playlist "Gathering songs" process never ends. I have to kill Shuttle in order to continue using it.
 
**Steps to reproduce:**
 
1. Go to Library
2. Go to "FOLDERS" tab
3. Navigate to SD card
4. Navigate to folder containing music files (for example my folder has 11 FLAC files)
5. Press three dots on a folder to open menu
6. Press "Add to playlist" > "New"
7. Enter playlist name
8. Press "SAVE"
 
**Expected outcome:**
 
Selected folder content is added to a new playlist quickly.
 
**Observations/Actual Result:**
 
 A dialog "Gathering songs..." is shown continuously, process never ends. Doesn't matter the number of music files that the folder contains.

**Comment by timusus** [comment]
The `0000-0000` looks suspicious in your folder path.

If you navigate back to `storage`, is there another path that leads to `Music`?

**Comment by DarkSkyGhost** [comment]
I think this is default mount point created by system: /storage/0000-0000

As far as I looked there is no other path leading to Music.

Same thing happens when I insert another SD card, for which system creates another folder (not 0000-0000).

Internal phone storage is mounted in /storage/emulated/0 and if I try to add files from internal storage, gathering songs process hangs too.

**Comment by timusus** [comment]
Interesting. Can you do anything else with the files? Also, are these files present in the regular media library (the artists/albums/songs tabs)?

**Comment by DarkSkyGhost** [comment]
Files are fully accessible from both stock and nonstock (FX) file managers - renaming, deleting and other modifications work.

And yes, there are files present in regular media library and adding them to the playlist works normally - I am able to play them.

**Comment by timusus** [comment]
OK, so it's just adding folders causing trouble? Definitely sounds like an app issue, I'll look into it.

**Comment by DarkSkyGhost** [comment]
Yeah, everything else works perfectly except adding folders to the playlist.

If you need any debug info, just let me know.

**Comment by Samelandslaget** [comment]
Any updates on this? I have the same issue on my Moto G5 Plus.

**Comment by timusus** [comment]
I can't reproduce this issue, so I don't have any updates at this stage.
